digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/SpeedLimitService.scala

```diff
@@ -336,8 +367,36 @@ class SpeedLimitService(eventbus: DigiroadEventBus, vvhClient: VVHClient, roadLi
     dao.updateExpiration(id, true, username)
 
     //Create New Asset copy by the old one with new value
-    dao.createSpeedLimit(speedLimit.createdBy.getOrElse(username), speedLimit.linkId, measures.getOrElse(Measures(speedLimit.startMeasure, speedLimit.endMeasure)),
-      SideCode.apply(sideCode.getOrElse(speedLimit.sideCode.value)), value, Some(speedLimit.vvhTimeStamp), speedLimit.createdDate, Some(username), Some(DateTime.now()), speedLimit.linkSource)
+    val newAssetId =
+      dao.createSpeedLimit(speedLimit.createdBy.getOrElse(username), speedLimit.linkId, measures.getOrElse(Measures(speedLimit.startMeasure, speedLimit.endMeasure)),
+        SideCode.apply(sideCode.getOrElse(speedLimit.sideCode.value)), value, Some(speedLimit.vvhTimeStamp), speedLimit.createdDate, Some(username), Some(DateTime.now()), speedLimit.linkSource)
+
+    existOnQualityErrorsList(id, newAssetId)
+    newAssetId
+  }
+
+  private def existOnQualityErrorsList(oldId: Long, newId: Option[Long] = None) = {
+    (inaccurateAssetDao.getInaccurateAssetById(oldId), newId) match {
+      case (Some(idOld), Some(idNew)) =>
+        inaccurateAssetDao.deleteInaccurateAssetById(idOld)
+        checkInaccurateSpeedLimit(newId)
+      case _ => None
+    }
+  }
+
+  private def checkInaccurateSpeedLimit(id: Option[Long] = None) = {
+    dao.getSpeedLimitLinksById(id.head).foreach { speedLimit =>
+
+      val roadLink = roadLinkServiceImplementation.getRoadLinksAndComplementariesFromVVH(Set(speedLimit.linkId), false)
+        .find(roadLink => roadLink.administrativeClass == State || roadLink.administrativeClass == Municipality)
```

Comment:
getRoadLinksAndComplementariesFromVVH(Set(speedLimit.linkId), false) could be replaced by getRoadLinkAndComplementaryFromVVH(speedLimit.linkId, false)